build/gulpfile.vscode.js

```diff
@@ -454,6 +454,22 @@ gulp.task(task.define(
 	)
 ));
 
+// {{SQL CARBON EDIT}} Localization gulp task, similar to vscode-translations-export but only for ADS only extensions.
```

Comment:
Is there some reason we couldn't use that task instead? Not sure why we need a special one for ADS-only extensions if this is actually doing basically the same thing

Reply:
The vscode-translations-export task only includes extensions that are not marked as external extensions rather than all extensions due to using packagelocalextensionstream in extensions.ts. This factors into the extensions that will be bundled with ADS or packaged separately. Since some of these extensions are not meant to be bundled with ADS, either a flag passed in to indicate the usage of the packagelocalextensionstream code or creating a new function based on it is justified.

Reply:
Ok, so I would instead clarify that this only packages EXTERNAL extensions. It's technically true that those all happen to be ADS extensions, but in this case I think it makes it a lot clearer. 

And then in that case can you not use https://github.com/microsoft/azuredatastudio/blob/main/build/lib/extensions.ts#L242? It'd be nice to not have yet another list of extensions we have to remember to update if we add a new extension.

Reply:
To clarify: it does not localize all of the External extensions as some of them do not have localization yet, there's also extensions that are bundled with ADS such as resource-deployment that aren't in vscode and therefore would need to be localized. This list of extensions in the code that I added therefore contains a mixture of bundled and external extensions that we currently are translating for, and that are not already translated in vscode. Only some extensions from the bundled and external extensions are used, and not all of them, hence the new list.

Reply:
Couldn't we either : 

1. Try to localize all extensions, and if it doesn't have any strings to localize then the output would just be empty (which shouldn't really impact anything)
2. Have some other way to detect if an extension should be localized, such as searching for a file that only exists if an extension should be localized (.i18n.json or package.nls.json?)

Reply:
Running this on all extensions causes a too many open files message when it happens though, I'll try to add a check for package.nls.json though.